secrets/secrets.go

```diff
@@ -298,3 +321,35 @@ func NewSecrets(r io.Reader) (*Secrets, error) {
 	}
 	return secrets, nil
 }
+
+func csiPathParser(dir fs.FS) (Document, error) {
+	secretsDocument := Document{
+		Secrets: make(map[string]GenericSecret),
+	}
+	err := fs.WalkDir(
+		dir,
+		".",
+		func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if d.IsDir() {
+				return nil
+			}
+			file, err := dir.Open(path)
+			if err != nil {
+				return err
+			}
+			defer file.Close()
+
+			var secretFile CSIFile
+			err = json.NewDecoder(file).Decode(&secretFile)
+			if err != nil {
+				return err
+			}
+			secretsDocument.Secrets[path] = secretFile.Secret
+			return nil
+		},
+	)
+	return secretsDocument, err
```

Comment:
Unless you explicitly need the return value in the error case, it's idiomatic to return `nil, err` in the multi-return error case. The error and the return value are generally considered to be mutually exclusive and this pattern makes the intended use explicit and lowers overall cognitive overhead. (In this case you'd also have to adjust the signature to return a `*Document`)